Mathlib/Analysis/Analytic/Prod.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2023 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+import Mathlib.Analysis.Analytic.Composition
+import Mathlib.Analysis.Analytic.Linear
+
+/-!
+# Analytic functions on and to product spaces
+-/
+
+noncomputable section
+
+open Topology Classical BigOperators NNReal Filter ENNReal
+
+open Function (uncurry)
+
+open Set Filter Asymptotics
+
+variable {𝕜 E F G H : Type*}
+
+variable [NontriviallyNormedField 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NormedAddCommGroup F]
+  [NormedSpace 𝕜 F] [NormedAddCommGroup G] [NormedSpace 𝕜 G] [NormedAddCommGroup H]
+  [NormedSpace 𝕜 H]
+
+/-- Products of analytic functions are analytic -/
+theorem AnalyticAt.prod {f : E → F} {g : E → G} {x : E} (fa : AnalyticAt 𝕜 f x)
+    (ga : AnalyticAt 𝕜 g x) : AnalyticAt 𝕜 (fun x ↦ (f x, g x)) x := by
+  rcases fa with ⟨p, pr, fp⟩
+  rcases ga with ⟨q, qr, gq⟩
+  set pq : FormalMultilinearSeries 𝕜 E (F × G) := fun n ↦ (p n).prod (q n)
+  have pqr : min pr qr ≤ pq.radius := by
+    apply ENNReal.le_of_forall_nnreal_lt; intro r rr
```

Comment:
Normally, we use a single tactic per line (except in one line proofs). Here, you could use 
```
    refine ENNReal.le_of_forall_nnreal_lt (fun r rr ↦ ?_)
```
to get the same effect as your two tactics.

Reply:
Will do in future (though apologies in advance as I expect to accidentally let a variety of them through as I do this a lot in the current code).  In any case, obsolete here as this function was already upstreamed by someone else.